pkg/kubelet/kubelet.go

```diff
@@ -875,7 +883,25 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		kubeDeps.Recorder,
 		volumepathhandler.NewBlockVolumePathHandler())
 
-	klet.backOff = flowcontrol.NewBackOff(backOffPeriod, MaxContainerBackOff)
+	max := MaxContainerBackOff
+	base := containerBackOffPeriod
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableKubeletCrashLoopBackoffMax) {
+		if kubeCfg.CrashLoopBackOff.MaxSeconds != nil {
+			max = time.Duration(*kubeCfg.CrashLoopBackOff.MaxSeconds)
+			if max < containerBackOffPeriod {
+				base = max
+			}
+		} else {
+			klog.InfoS("EnableKubeletCrashLoopBackOffMax feature gate enabled, but crashloopbackoff.maxSeconds KubeletConfig not set. Using default CrashLoopBackOff maximum backoff.")
```

Comment:
If the `EnableKubeletCrashLoopBackOffMax` FeatureGate is enabled by default, this log message will always be written when kubelet starts unless the cluster administrator explicitly sets `crashloopbackoff.maxSeconds`. I think it makes sense to log a message when the FeatureGate is disabled and `crashloopbackoff.maxSeconds` is specified for kubelet. However, since there could be cases where the FeatureGate is enabled but `crashloopbackoff.maxSeconds` is not set, I wondered if this message is really necessary at least after it has been enabled by default.

Reply:
The initial thought was to increase the places at which one could detect which nodes in a cluster are configured with different crashloopbackoff maximums -- which I think is still valuable even after it's a default (maybe with different wording or just a startup log that reports the current maximum). FWIW, if this is an anti-pattern, I'm not actually opposed to removing it, I may be oversensitive because I spend a lot of time in an oncall rotation that invovles reading a lot of kubelet logs ;)

Another option, is removing this logline along with the feature gate when the time comes that it is on by default. I can put a specific comment here, tied a comment against the feature gate so it's clear.

Reply:
> I may be oversensitive because I spend a lot of time in an oncall rotation that invovles reading a lot of kubelet logs ;)

I see; from that perspective, it does seem better to have it. However, since the configuration used by kubelet can be checked via the API, I'd like to hear the opinions of other reviewers on what would be optimal.

```sh
NODE_NAME=
kubectl get --raw "/api/v1/nodes/${NODE_NAME}/proxy/configz" | jq
```

I think there wouldn't be an issue as long as we update the part of the message that mentions the FeatureGate after the FeatureGate is removed. Adjusting the log level might also be a good idea.